src/js/tasks/dateDisplayFormatter.js

```diff
@@ -0,0 +1,231 @@
+Date.prototype.fromNow = function () {
+  let date = this;
+  let now = new Date();
+
+  let dYears = now.getFullYear() - date.getFullYear();
+  let dMonths = now.getMonth() - date.getMonth();
+  let dDays = now.getDate() - date.getDate();
+
+  if (dDays < 0) {
+    dMonths--;
+    dDays += new Date(date.getFullYear(), date.getMonth(), 0).getDate();
+  }
+
+  if (dMonths < 0) {
+    dYears--;
+    dMonths += 12;
+  }
+
+  let message = "";
+  if (dYears > 0) {
+    message += `${dYears} years `;
+  }
+
+  if (dMonths > 0) {
+    message += `${dMonths} months `;
+  }
+
+  if (dDays > 0) {
+    message += `${dDays} days `;
+  }
+
+  if (message.length > 0) {
+    message += "ago";
+  } else {
+    message = "Today";
+  }
+
+  return message;
+};
+
+const dateDisplayFormatter = (() => {
+  const monthNamesLong = [
+    "January",
+    "February",
+    "March",
+    "April",
+    "May",
+    "June",
+    "July",
+    "August",
+    "September",
+    "October",
+    "November",
+    "December",
+  ];
+
+  const monthNamesNarrow = [
+    "Jan",
+    "Feb",
+    "Mar",
+    "Apr",
+    "May",
+    "Jun",
+    "Jul",
+    "Aug",
+    "Sep",
+    "Oct",
+    "Nov",
+    "Dec",
+  ];
+
+  function getMonthPattern(format) {
+    if (format.indexOf("MMMM") != -1) {
+      return "MMMM";
+    } else if (format.indexOf("MMM") != -1) {
+      return "MMM";
+    } else if (format.indexOf("MM") != -1) {
+      return "MM";
+    } else {
+      return null;
```

Comment:
this null is showing on th UI later, it is better to show some validation error on the demo page